src/redturtle/prenotazioni/browser/prenotazioni_context_state.py

```diff
@@ -286,17 +297,42 @@ def is_before_allowed_period(self, day, bypass_user_restrictions=False):
     @memoize
     def is_valid_day(self, day, bypass_user_restrictions=False):
         """Returns True if the day is valid"""
```

Comment:
List the cases for which control is bypassed and those for which it is not

Reply:
bypass_user_restrictions was not added in this pr, it existed before: as for this pr bypass_user_restrictions is true when user is manager and "restrict user by validity days" flag is off. the other cases must be studied

Reply:
ok

Reply:
Yes, but before the code was really simple, now there is much more complexity. Before it wasn't necessary, now it is

Reply:
added doc